svm-decode/src/lib.rs

```diff
@@ -184,6 +216,102 @@ pub fn decode_instructions(
     Ok(batch)
 }
 
+pub fn svm_decode_logs(
+    signature: LogSignature,
+    batch: &RecordBatch,
+    allow_decode_fail: bool,
+) -> Result<RecordBatch> {
+    let message_col = batch.column_by_name("message").unwrap();
+    let data = message_col.as_any().downcast_ref::<StringArray>().unwrap();
+
+    decode_logs(signature, data, allow_decode_fail)
+}
```

Comment:
A simple way for the decoding to fail is to run the decode_logs function when the message is of kind 'log' or 'other' and not 'data'. (Message is a simple text and not serialized data.)

Logs contains the kind column, but in other to skip the rows with `kind != 'data'`, I would need to downcast it and pass it along to the decode_logs functions, so there I could test it as well.
I didn't like this alternative.
So, for now, if you have other kinds in the data, you need to set allow_decode_fail to true.

Reply:
Can do this filtering in `cherry` step implementation, would be ok to add it here as well